src/scenes/Question/scenes/Answer/components/Sources/index.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import uuid from 'uuid/v4'
+
+import Source from '../Source'
+
+import Button from 'react-toolbox/lib/button/Button'
+
+class Sources extends Component {
+  addSource () {
+    const { sources, handleChange } = this.props
+    handleChange(sources.concat([{ id: uuid(), label: '', url: '' }]))
+  }
+
+  changeLabel (id, label) {
+    const { sources, handleChange } = this.props
+    handleChange(
```

Comment:
you could currify handleChange to make it reusable:

```js
const changeSomething = (id, changes) => sources => {
  handleChange(
    sources.map(s => {
      if (s.id !== id) return s
      return { ...s, ...changes }
    })
  )
}
```

```js
changeLabel (id, label) {
  const { sources, handleChange } = this.props
  changeSomething(id, { label })(sources)
}
```

What do you think ?

Reply:
I like that !

Reply:
In the end, I don't think I need a currifying method, I can just do this:
```js
  changeAttribute = (id, attributes) => {
    const { sources, handleChange } = this.props
    handleChange(
      sources.map(s => {
        if (s.id !== id) return s
        return { ...s, ...attributes }
      })
    )
  }

  changeLabel(id, label) {
    this.changeAttribute(id, { label })
  }
```

Reply:
👍